apis/Google.Longrunning/Google.Longrunning/Operation.cs

```diff
@@ -0,0 +1,175 @@
+        /// </summary>
+        public Operation Proto { get; }
+        public OperationsClient Client { get; }
+
+        public Operation(Operation proto, OperationsClient client)
+        {
+            Proto = GaxPreconditions.CheckNotNull(proto, nameof(proto));
+            Client = GaxPreconditions.CheckNotNull(client, nameof(proto));
+        }
+
+        /// <summary>
+        /// Returns the name of the operation, which can be persisted and used to poll for the latest
+        /// results at a later time or in a different program.
+        /// </summary>
+        /// <remarks>
+        /// Only the in-memory representation of the operation (this object) is consulted for its state.
+        /// </remarks>
+        public string Name => Proto.Name;
+
+        /// <summary>
+        /// Whether the operation has completed, where "complete" can include "failed".
+        /// </summary>
+        /// <remarks>
+        /// Only the in-memory representation of the operation (this object) is consulted for its state.
+        /// </remarks>
+        public bool IsCompleted => Proto.Done;
+
+        /// <summary>
+        /// Whether the operation has completed with a failure.
+        /// </summary>
+        public bool IsFaulted => Proto.ResultCase == Operation.ResultOneofCase.Error;
+
+        // TODO: An Exception property? Would probably want to use a Lazy<OperationFailedException> to avoid creating
+        // a new one each time. Hmm.
+
+        /// <summary>
+        /// Retrieves the result of the operation, throwing an exception if the operation failed or hasn't completed.
+        /// </summary>
+        /// <remarks>
+        /// Only the in-memory representation of the operation (this object) is consulted for its state.
+        /// </remarks>
+        /// <exception cref="OperationFailedException">The operation completed with an error.</exception>
+        /// <exception cref="InvalidOperationException">The operation has not completed yet.</exception>
+        public T Result
+        {
+            get
+            {
+                switch (Proto.ResultCase)
+                {
+                    case Operation.ResultOneofCase.Error:
+                        var error = Proto.Error;
+                        // TODO: Construct a new exception type?
+                        throw new RpcException(new Status((StatusCode) error.Code, error.Message));
+                    case Operation.ResultOneofCase.Response:
+                        return Proto.Response.Unpack<T>();
+                    default:
+                        throw new InvalidOperationException();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Polls the operation until it is complete, returning the completed operation.
+        /// </summary>
+        /// <remarks>
+        /// If this object represents a completed operation
+        /// </remarks>
+        public Operation<T> PollUntilCompleted(PollSettings pollSettings = null)
```

Comment:
Could do, or we could have an extra CallSettings parameter - I *suspect* the latter makes more sense.